sim/core/flags.go

```diff
@@ -176,36 +176,35 @@ func (se SpellFlag) Matches(other SpellFlag) bool {
-	SpellFlagNoMetrics                                      // Disables metrics for a spell.
-	SpellFlagNoLogs                                         // Disables logs for a spell.
-	SpellFlagAPL                                            // Indicates this spell can be used from an APL rotation.
-	SpellFlagMCD                                            // Indicates this spell is a MajorCooldown.
-	SpellFlagReactive                                       // Allows a spell flagged as an MCD to be cast off-GCD. Used for instant cast defensive CDs.
-	SpellFlagNoOnDamageDealt                                // Disables OnSpellHitDealt and OnPeriodicDamageDealt aura callbacks for this spell.
-	SpellFlagPrepullOnly                                    // Indicates this spell should only be used during prepull. Not enforced, just a signal for the APL UI.
-	SpellFlagEncounterOnly                                  // Indicates this spell should only be used during the encounter (not prepull). Not enforced, just a signal for the APL UI.
-	SpellFlagPotion                                         // Indicates this spell is a potion spell.
-	SpellFlagPrepullPotion                                  // Indicates this spell is the prepull potion.
-	SpellFlagCombatPotion                                   // Indicates this spell is the combat potion.
-	SpellFlagNoSpellMods                                    // Indicates that no spell mods should be applied to this spell
-	SpellFlagCanCastWhileMoving                             // Allows the cast to be casted while moving
-	SpellFlagPassiveSpell                                   // Indicates this spell is applied/cast as a result of another spell
-	SpellFlagSupressDoTApply                                // If present this spell will not apply dots (Used for DTR dot supression)
-	SpellFlagSwapped                                        // Indicates that this spell is not useable because it is from a currently swapped item
+	SpellFlagNone                    SpellFlag = 0
+	SpellFlagIgnoreResists           SpellFlag = 1 << iota // skip spell resist/armor
+	SpellFlagIgnoreTargetModifiers                         // skip target damage modifiers
+	SpellFlagIgnoreAttackerModifiers                       // skip attacker damage modifiers
+	SpellFlagApplyArmorReduction                           // Forces damage reduction from armor to apply, even if it otherwise wouldn't.
+	SpellFlagCannotBeDodged                                // Ignores dodge in physical hit rolls
+	SpellFlagBinary                                        // Does not do partial resists and could need a different hit roll.
+	SpellFlagChanneled                                     // Spell is channeled
+	SpellFlagDisease                                       // Spell is categorized as disease
+	SpellFlagHauntSE                                       // Spell benefits from haunt/SE effects
```

Comment:
I think we should remove `SpellFlagHauntSE` as a core flag as well, Warlock code can always use one of the `AgentReserved` flags if needed rather than defining a general core constant for class-specific logic.